Add GetBlock Tutorial to the documentation section (#1385)
maticnetwork/matic-docs: docs/develop/getblock.md

## What you will do

In this tutorial, you will:

1. Get started on GetBlock and access test nodes.
2. Create a wallet address with Metamask.
3.Add test tokens to your wallet.
4. Compile and deploy a smart contract using Remix
5. Check the status of your smart contract.

### Preparatory Steps

1. The first step is to [sign up with Getblock](https://account.getblock.io/sign-in) via wallet or email via 

  ![img](https://storage.getblock.io/web/blog/article-images/img1+(2).png)

2. Create an endpoint for Polygon testnet (Mumbai).

  ![img](https://storage.getblock.io/web/blog/article-images/img2+(1).png)

3. Then enter Metamask and choose to connect a network manually, connect your Metamask to the Mumbai Testnet adding your GetBlock endpoint in the **RPC URL** line

  ![img](https://storage.getblock.io/web/blog/article-images/imga3.png)

If you don't have any tokens in your account, you can request some from the Mumbai faucet through the [faucet website](https://chat.openai.com/develop/tools/matic-faucet.md).

## Smart Contract Development

> - Go to the [official Remix](https://remix.ethereum.org/) website. Remix is a Web-based IDE for Ethereum smart contract development in Solidity. 

Reviewer: ```suggestion
1. Visit the [official Remix website](https://remix.ethereum.org/). Remix is a Web-based IDE for Ethereum smart contract development in Solidity. 
```

> - Click on the "Create a new file" button located on the left side of the page.

Reviewer: ```suggestion
2. Click on the "Create a new file" button located on the left side of the page.
```

> - Select “Storage.sol” from Remix as an example of a contract.

Reviewer: ```suggestion
3. Select “Storage.sol” from Remix as an example of a contract.
```

  ![img](https://storage.getblock.io/web/blog/article-images/imga4.png)

> - Proceed by deploying the smart contract by clicking on the “Deploy & Run Transactions” tab.

Reviewer: ```suggestion
4. Deploy the smart contract by clicking on the “Deploy & Run Transactions” tab.
```

> - In Remix, click on the “Deploy & Run Transactions” tab located on the left-hand side of the screen. 

Reviewer: ```suggestion
5. Click on the “Deploy & Run Transactions” tab in Remix, located on the left-hand side of the screen. 
```

> - In the “Contract” dropdown menu, select the name of the smart contract you want to deploy and click on the “Deploy” button to deploy the smart contract.

Reviewer: ```suggestion
6. In the “Contract” dropdown menu, select the name of the smart contract you want to deploy and click on the “Deploy” button to deploy the smart contract.
```

A Metamask window will pop up asking you to confirm the transaction. Click on the “Confirm” button to proceed.Wait for the transaction to be confirmed on the Polygon network. 

> - You can track the status of the transaction in the Metamask window or by checking the transaction hash on the Polygon explorer.

Reviewer: ```suggestion
> You can track the status of the transaction in the Metamask window or by checking the transaction hash on the Polygon explorer.
```

> - Once the transaction has been confirmed, you should see a green checkmark and a message indicating that the contract has been deployed successfully.

Reviewer: ```suggestion
Once the transaction has been confirmed, you should see a green checkmark and a message indicating that the contract has been deployed successfully.
```

  ![img](https://storage.getblock.io/web/blog/article-images/imga5.png)

> Locate the storage variable you want to test.

Reviewer: ```suggestion
7. Locate the storage variable you want to test.
```

> - Choose "Injected provider" as the provider and select "Metamask" as the wallet. 

Reviewer: ```suggestion
8. Choose "Injected provider" as the provider and select "Metamask" as the wallet. 
```

 ![img](https://storage.getblock.io/web/blog/article-images/imga6.png)
  

> - Check that your smart contract has been created on the network by looking at the Remix output panel.

Reviewer: ```suggestion
> Check that your smart contract has been created on the network by looking at the Remix output panel.
```

> - Click on the "Polygon (Matic) Mainnet" dropdown menu located at the top right corner of the page and select "Mumbai Testnet" from the list of available networks. Then enter the address of your smart contract in the search bar located at the top of the page.

Reviewer: ```suggestion
9. Click on the "Polygon (Matic) Mainnet" dropdown menu located at the top right corner of the page and select "Mumbai Testnet" from the list of available networks. Then enter the address of your smart contract in the search bar located at the top of the page.
```

> - Click on the search button or hit enter.

Reviewer: ```suggestion
10. Click on the search button or hit enter.
```

  
 ![img](https://storage.getblock.io/web/blog/article-images/img6+(1).png)

Smart contract URL: https://mumbai.polygonscan.com/address/0xa4d552dce0e26564ec1737638705584ce9ed351b

If your smart contract has been successfully deployed, you should see the contract details including its address, balance, and transaction history. Congratulations, you have successfully deployed a smart contract on the Mumbai Testnet! 

## Testing and Verification

- Once you have deployed your smart contract on the Polygon test network using GetBlock, it is important to test and verify its functionality to ensure that it works as intended. Here are some steps you can take for testing and verification:

> - Open the smart contract in Remix and locate the storage variable you want to test. In this example, we will use a variable called "storage" that initially has a value of 12. 

Reviewer: ```suggestion
1. Open the smart contract in Remix and locate the storage variable you want to test. In this example, we will use a variable called "storage" that initially has a value of 12."
```

 ![img](https://storage.getblock.io/web/blog/article-images/img7+(1).png)

> - Find the "Store" field and enter the number "14". This will change the value of the "storage" variable from 12 to 14.

Reviewer: ```suggestion
2. Change the value of the variable to 14 by entering "14" in the "Store" field and saving the value on the network after confirming the operation in Metamask.
```

> - Click on the "Transact" button to confirm the transaction in Metamask.

Reviewer: ```suggestion
3. Click on the "Transact" button to confirm the transaction in Metamask.
```

 ![img](https://storage.getblock.io/web/blog/article-images/imga10.png)

> - Wait for the transaction to be added to the network block. Once the transaction is confirmed, go to the "Read Contract" section under "Deployed Contracts".

Reviewer: ```suggestion
4. Wait for the transaction to be added to a network block. Once the transaction is confirmed, navigate to the "Read Contract" section under "Deployed Contracts".
```

> - Find the "storage" variable and click on the "Read" button. The new value of the variable should now be displayed as 14.

Reviewer: ```suggestion
4. Find the "storage" variable and click on the "Read" button. The new value of the variable should now be displayed as "14".
```

title: Deploy a Smart Contract on Polygon using GetBlock
sidebar_label: Using GetBlock
description: Guide to deploy smart contracts using GetBlock
keywords:
  - docs
  - matic
  - polygon
  - getblock
  - create smart contract
  - deploy on polygon
  - verify your smart contract
image: https://wiki.polygon.technology/img/polygon-logo.png
---

Polygon node connection is an advantage for any dApp whether it's an on-chain game or trading bots. Polygon API endpoint by GetBlock is an instrument of choice for many early-stage teams. This is an easy step-by-step tutorial on how to create and deploy an ERC-20 smart contract on the Polygon Mumbai testnet using [Metamask](https://metamask.io/), Remix, and [GetBlock](https://getblock.io/nodes/matic/).

You will learn how to use GetBlock to create and deploy a smart contract on Mumbai Testnet using [GetBlock](https://getblock.io/) endpoints. 

## What you will do

Following the tutorial, you will:
1. Get started on GetBlock and access GetBlock test nodes
2. Create a wallet address with Metamask
3. Add balance to the wallet (using test tokens)
4. Use Remix to compile and deploy a Smart Contract
5. Check contract status

### Preparatory Steps

> 1. The first step is to [sign up with Getblock](https://account.getblock.io/sign-in) via wallet or email

Reviewer: ```suggestion
1. Sign up with Getblock by visiting the [official website](https://account.getblock.io/sign-in) and creating an account using your wallet or email.
```

  ![img](https://storage.getblock.io/web/blog/article-images/guide1.png)

> 2. Create an endpoint for Mumbai Testnet

Reviewer: I'd create a list that includes step-by-step instructions, such as:

```
Create an endpoint for the Mumbai Testnet on Getblock by following these steps:

- Go to the Getblock website and log in to your account.
- Click on "API Keys" in the left-hand menu.
- Click on "Create API Key" and select "Ethereum" as the blockchain.
- Choose "Mumbai Testnet" as the network and give your endpoint a name.
- Click "Create."
- Once your endpoint is created, copy the RPC URL.
```

  ![img](https://storage.getblock.io/web/blog/article-images/guide2.png)

>   Then enter Metamask and choose to connect a network manually, connect your Metamask to the Mumbai Testnet adding your GetBlock endpoint in the **RPC URL** line

Reviewer: Same idea here, such as:

```
Connect Metamask to the Mumbai Testnet by following these steps:

- Open Metamask and click on the network dropdown.
- Click on "Custom RPC" at the bottom of the list.
- In the "RPC URL" field, paste the URL of your Getblock endpoint.
- Click "Save."

```

  ![img](https://storage.getblock.io/web/blog/article-images/guide3.png)
  ![img](https://storage.getblock.io/web/blog/article-images/guide4.png)

> If you have zero balance in your account, you can request tokens from the [Mumbai faucet](/develop/tools/matic-faucet.md).

Reviewer: ```suggestion
If you don't have any tokens in your account, you can request some from the Mumbai faucet through the [faucet website](https://chat.openai.com/develop/tools/matic-faucet.md).
```

## Smart Contract Development

> Go to the [official Remix](https://remix.ethereum.org/) website. Remix is a Web-based IDE for Ethereum smart contract development in Solidity. 

Reviewer: ```suggestion
1. Go to the [official Remix](https://remix.ethereum.org/) website, which is a Web-based IDE for Ethereum smart contract development in Solidity.
```

> 1. Let's take `Storage.sol` from Remix as our example of a smart contract.

Reviewer: ```suggestion
2. Open the "Storage.sol" file in Remix, which we will use as an example for our smart contract.
```

  ![img](https://storage.getblock.io/web/blog/article-images/guide5.png)

> 2. Deploy the smart contract.

Reviewer: ```suggestion
3. Deploy the smart contract by clicking on the "Deploy & run transactions" icon in the left-hand menu.
```

  ![img](https://storage.getblock.io/web/blog/article-images/guide6.png)

> 3. Choose the Injected provider as **Metamask**. 

Reviewer: ```suggestion
4. Choose "Injected provider" as the provider and select "Metamask" as the wallet.
```

 ![img](https://storage.getblock.io/web/blog/article-images/guide7.png)
  

>   Check that our smart contract has been created on the network.

Reviewer: ```suggestion
5. Check that your smart contract has been created on the network by looking at the Remix output panel.
```

  
 ![img](https://storage.getblock.io/web/blog/article-images/guide8.png)

Smart contract URL: https://mumbai.polygonscan.com/address/0xa4d552dce0e26564ec1737638705584ce9ed351b

> All done! Congratulations on deploying your smart contract.

Reviewer: ```suggestion
Congratulations, you have successfully deployed a smart contract on the Mumbai Testnet! 
```

## Testing and Verification

Once you have deployed your smart contract on the Polygon test network using GetBlock, it is important to test and verify its functionality to ensure that it works as intended. Here are some steps you can take for testing and verification:

> 

Reviewer: Would also suggest transforming this section into more explicit steps, starting with something along the lines of:

```
1. Open the smart contract in Remix and locate the storage variable you want to test. In this example, we will use a variable called "storage" that initially has a value of 12."
```

Striving to create clear + direct and actionable steps.

```
2. Change the value of the variable to 14 by entering "14" in the "Store" field and saving the value on the network after confirming the operation in Metamask.
```

And so on.

1. Let's take `Storage.sol` from Remix as our example of a smart contract.

  ![img](https://i.ibb.co/7JxRcS5/unnamed-2.png)
  ![img](https://drive.google.com/file/d/12PEfBR7Tju9ReVYVtipSn3iR_rndMxEs/view?usp=sharing)

2. Deploy the smart contract.

>   ![img](https://i.ibb.co/FHQvbb0/unnamed-3.png)

Reviewer 2: I appreciate for all your good work I'm becoming better understanding I even con
